plugins/openapi/pkg/openapiclient/openapi.go

```diff
@@ -0,0 +1,181 @@
+package openapiclient
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/pkg/errors"
+)
+
+const (
+	Base64    = "base64"
+	Canonical = "canonical"
+)
+
+type OpenAPIClient struct {
+	docs     []*openapi3.T
+	idFormat string
+}
+
+type API struct {
+	Type        string `json:"type"`
+	Service     string `json:"service"`
+	Method      string `json:"method"`
+	Path        string `json:"path"`
+	DisplayName string `json:"displayName"`
+	ID          string `json:"id"`
+	ServiceID   string `json:"serviceID"`
+}
+
+type Service struct {
+	Type        string `json:"type"`
+	DisplayName string `json:"displayName"`
+	ID          string `json:"id"`
+}
+
+func NewOpenAPIClient(directory, specurl, idFormat string) (*OpenAPIClient, error) {
```

Comment:
I would rename `specurl` into `specURL`; the function signature is more important in terms of clarity than the body.